workspaces/bi/bi-extension/src/test/e2e-playwright-tests/api-services/graphqlUtils.ts

```diff
@@ -0,0 +1,146 @@
+    },
+};
+
+/**
+ * Utility to add a GraphQL operation (mutation, subscription, etc.)
+ * @param artifactWebView - The Playwright frame/locator for the webview
+ * @param operationType - 'mutation' | 'subscription' | 'query'
+ * @param name - The name to use for the operation
+ * @param fieldType - The type to use for the field (e.g., 'boolean', 'float', etc.)
+ */
+export async function addGraphQLOperation(artifactWebView: Frame, operationType: string, name: string, fieldType: string) {
+    const addBtnTestId = `graphql-add-${operationType}-btn`;
+    await artifactWebView.getByTestId(addBtnTestId).waitFor({ state: 'visible', timeout: 10000 });
+    const addBtn = artifactWebView.getByTestId(addBtnTestId);
+    await addBtn.click();
+
+    const fieldNameBox = artifactWebView.getByRole('textbox', { name: /Field Name/i });
+    await fieldNameBox.waitFor({ state: 'visible', timeout: 10000 });
+    await fieldNameBox.fill(name);
+
+    const fieldTypeBox = artifactWebView.getByRole('textbox', { name: /Field Type/i });
+    await fieldTypeBox.waitFor({ state: 'visible', timeout: 10000 });
+    // await fieldTypeBox.click();
+    await fieldTypeBox.fill(fieldType);
+
+    // Wait a short moment to allow UI to register the value
+    await page.page.waitForTimeout(5000);
+    const fieldDefaultCompletion = artifactWebView.getByTestId('add-type-completion');
+    await fieldDefaultCompletion.waitFor({ state: 'visible', timeout: 10000 });
+
+    if (fieldDefaultCompletion.isVisible()) {
+        await fieldTypeBox.press('Escape');
+    }
+
+    const saveBtn = artifactWebView.getByRole('button', { name: /Save/i });
+    await saveBtn.waitFor({ state: 'visible', timeout: 10000 });
+    await saveBtn.click();
+}
+
+/**
+ * Click a button in the artifact webview by test id
+ * @param artifactWebView - The Playwright frame/locator for the webview
+ * @param testId - The test id of the button to click
+ */
+export async function clickButtonByTestId(artifactWebView: Frame, testId: string) {
+    const button = artifactWebView.getByTestId(testId);
+    await button.waitFor({ state: 'visible', timeout: 10000 });
+    await button.click();
+}
+
+/**
+ * Create an output object type in the GraphQL service
+ * @param artifactWebView - The Playwright frame/locator for the webview
+ */
+export async function addOutputObject(artifactWebView: Frame) {
+    const createFromScratchTab = artifactWebView.getByTestId('create-from-scratch-tab');
+    await artifactWebView.getByRole('textbox', { name: 'Field Type' }).click();
+    await artifactWebView.getByText('Create New Type').click();
+    await artifactWebView.getByTestId('type-kind-dropdown').locator('svg').click();
+    await artifactWebView.getByRole('option', { name: 'Object' }).click();
+    await createFromScratchTab.getByRole('textbox', { name: 'Name' }).fill(TEST_DATA.mutation[0].outputType);
+    await artifactWebView.getByTestId('type-create-save').getByRole('button', { name: 'Save' }).click();
+}
```

Comment:
Shall we use the same utils that we have used in the TypeEditor. since this object is similar to creating a service class in TypeEditor. same applies on inputObject creation as well